docker/checkout.bash

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+#  Copyright (C) 2021 LEIDOS.
+# 
+#  Licensed under the Apache License, Version 2.0 (the "License"); you may not
+#  use this file except in compliance with the License. You may obtain a copy of
+#  the License at
+# 
+#  http://www.apache.org/licenses/LICENSE-2.0
+# 
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#  License for the specific language governing permissions and limitations under
+#  the License.
+
+# CARMA packages checkout script
+# Optional argument to set the root checkout directory with no ending '/' default is '~'
+
+set -exo pipefail
+
+dir=~
+while [[ $# -gt 0 ]]; do
+      arg="$1"
+      case $arg in
+            -d|--develop)
+                  BRANCH=develop
+                  shift
+            ;;
+            -r|--root)
+                  dir=$2
+                  shift
+                  shift
+            ;;
+      esac
+done
+
+if [[ "$BRANCH" = "foxy/develop" ]]; then
+      sudo git clone https://github.com/usdot-fhwa-stol/carma-msgs.git ${dir}/src/CARMAMsgs --branch $BRANCH
+      sudo git clone https://github.com/novatel/novatel_oem7_driver.git --branch ros2-dev
```

Comment:
Identify the current commit and update this to change to that exact commit. Pulling in external repos by branch name has caused problems for us, so we want to target specific commits for repos we don't manage.